### PR TITLE
Refactor: namespace everything

### PR DESCRIPTION
**Pull request by bpolaszek**
Depends on #12 by @simPod, following discussion on #8.

- `iterable_*` functions are no longer in the global namespace
- unrelevant checks for PHP5 have been dropped
- added PHP7 type-hints
- removed useless PHPDoc comments.

### CODE REVIEW

File: src/IterableObject.php [hunk 1]
```diff
@@ -18,113 +19,50 @@ final class IterableObject implements IteratorAggregate
...
-                sprintf('Expected callable, got %s', is_object($map) ? get_class($map) : gettype($map))
-            );
-        }
-
-        $this->iterable = $iterable;
-        $this->filter = $filter;
-        $this->map = $map;
+        $this->iterable = $iterable ?? new EmptyIterator();
+        $this->filterFn = $filter;
+        $this->mapFn = $map;
```
Comment (simPod) [inline src/IterableObject.php:33]: the names differ here

File: src/iterable-functions.php [hunk 2]
```diff
@@ -1,174 +1,108 @@
...
+
+use ArrayIterator;
+use CallbackFilterIterator;
+use IteratorIterator;
+use Traversable;
+
+/**
+ * Maps a callable to an iterable.
+ *
+ * @param array|Traversable $iterable
```
Comment (simPod) [inline src/iterable-functions.php:14]: We can use `iterable` already

**Review by simPod — APPROVED** [review]

File: src/iterable-functions.php (hunk 2 above)
Comment (bpolaszek) [reply src/iterable-functions.php:14]: IDE yells on line 23 if so, as it considers array_map could get an `iterable` (hence a `Traversable`).

File: src/IterableObject.php (hunk 1 above)
Comment (bpolaszek) [reply src/IterableObject.php:33]: Yes, I renamed internal props, as IDE complained properties had similar names than methods.

File: src/iterable-functions.php (hunk 2 above)
Comment (simPod) [reply src/iterable-functions.php:14]: I have gone forward with it, can solve later. Nvm

**Review by simPod — APPROVED** [review]

**Pull request merged**